sources/unstructured_data/inbox/helpers.py

```diff
@@ -0,0 +1,65 @@
+import email
+import imaplib
+from email.message import Message
+from typing import Any, Dict, Optional
+
+from dlt.common import pendulum
+
+
+def extract_email_info(msg: Message, body: bool = False) -> Dict[str, Any]:
+    email_data = dict(msg)
+    email_data["Date"] = pendulum.parse(msg["Date"], strict=False)
+    email_data["content_type"] = msg.get_content_type()
+    if body:
+        email_data["body"] = get_email_body(msg)
+
+    return {
+        k: v for k, v in email_data.items() if not k.startswith(("X-", "ARC-", "DKIM-"))
+    }
+
+
+def get_message_obj(client: imaplib.IMAP4_SSL, message_uid: str) -> Optional[Message]:
+    client.select()
+
+    status, data = client.uid("fetch", message_uid, "(RFC822)")
+    msg = None
+    if status == "OK":
+        raw_email = data[0]
+        if raw_email:
+            raw_email = data[0][1]
+            msg = email.message_from_bytes(raw_email)
+
+    return msg
+
+
+def get_internal_data(client: imaplib.IMAP4_SSL, message_uid: str) -> Optional[Any]:
+    client.select()
+    status, data = client.uid("fetch", message_uid, "(INTERNALDATE)")
+    date = None
+    if status == "OK":
+        str_datetime = data[0].decode().split('"')[1]
```

Comment:
https://docs.python.org/3/library/imaplib.html#imaplib.Internaldate2tuple

parse with this but then convert to pendulum datetime (this will create a local datetime without timezone)